integration/e2e/bench_write_batch_test.go

```diff
@@ -0,0 +1,143 @@
+		var network *nwo.Network
+		var ordererRunner *ginkgomon.Runner
+		var ordererProcess, peerProcess ifrit.Process
+
+		BeforeEach(func() {
+			network = nwo.New(nwo.BasicEtcdRaft(), tempDir, client, StartPort(), components)
+
+			// Generate config and bootstrap the network
+			network.GenerateConfigTree()
+			network.Bootstrap()
+
+			// Start all the fabric processes
+			ordererRunner, ordererProcess, peerProcess = network.StartSingleOrdererNetwork("orderer")
+		})
+
+		AfterEach(func() {
+			if ordererProcess != nil {
+				ordererProcess.Signal(syscall.SIGTERM)
+				Eventually(ordererProcess.Wait(), network.EventuallyTimeout).Should(Receive())
+			}
+
+			if peerProcess != nil {
+				peerProcess.Signal(syscall.SIGTERM)
+				Eventually(peerProcess.Wait(), network.EventuallyTimeout).Should(Receive())
+			}
+
+			network.Cleanup()
+		})
+
+		It("deploys and executes experiment bench", func() {
+			orderer := network.Orderer("orderer")
+			channelparticipation.JoinOrdererJoinPeersAppChannel(network, "testchannel", orderer, ordererRunner)
+			peer := network.Peer("Org1", "peer0")
+
+			chaincode := nwo.Chaincode{
+				Name:            "mycc",
+				Version:         "0.0",
+				Path:            "github.com/hyperledger/fabric/integration/chaincode/multi/cmd",
+				Lang:            "golang",
+				PackageFile:     filepath.Join(tempDir, "multi.tar.gz"),
+				Ctor:            `{"Args":["init"]}`,
+				SignaturePolicy: `AND ('Org1MSP.member','Org2MSP.member')`,
+				Sequence:        "1",
+				InitRequired:    true,
+				Label:           "my_multi_operations_chaincode",
+			}
+
+			network.VerifyMembership(network.PeersWithChannel("testchannel"), "testchannel")
+
+			nwo.EnableCapabilities(
+				network,
+				"testchannel",
+				"Application", "V2_0",
+				orderer,
+				network.PeersWithChannel("testchannel")...,
+			)
+			nwo.DeployChaincode(network, "testchannel", orderer, chaincode)
+
+			By("run invoke where only put state")
+			experiment := gmeasure.NewExperiment("Only PUT State " + desc)
+			AddReportEntry(experiment.Name, experiment)
+
+			experiment.SampleDuration("invoke N-5 cycle-1", func(idx int) {
+				RunInvokeOnlyPutState(network, orderer, peer, "invoke", useWriteBatch, 1)
+			}, gmeasure.SamplingConfig{N: 5})
+
+			experiment.SampleDuration("invoke N-10 cycle-1000", func(idx int) {
+				RunInvokeOnlyPutState(network, orderer, peer, "invoke", useWriteBatch, 1000)
+			}, gmeasure.SamplingConfig{N: 10})
+
+			experiment.SampleDuration("invoke N-10 cycle-10000", func(idx int) {
+				RunInvokeOnlyPutState(network, orderer, peer, "invoke", useWriteBatch, 10000)
+			}, gmeasure.SamplingConfig{N: 10})
+		})
+	},
+		Entry("without write batch", "without write batch", false),
+		Entry("with write batch", "with write batch", true),
+	)
+})
+
+func RunInvokeOnlyPutState(n *nwo.Network, orderer *nwo.Orderer, peer *nwo.Peer, fn string, startBatch bool, initial int) {
```

Comment:
Can we use a more expressive variable name than `initial`? It is not obvious what this is...

And shouldn't `startBatch` really be called `useWriteBatch` to be consistent?

Reply:
No. In this case it is passing a sign that the chaincode should call startBatch or not.

Reply:
I can do these tests

| useWriteBatch (on peer) | StartWriteBatch (on chaincode) |
| --- | --- |
| true | false |
| false | true |
| true | true |

Reply:
done